cpp/dolfinx/geometry/utils.cpp

```diff
@@ -279,57 +248,34 @@ double geometry::compute_squared_distance_bbox(
          + (d1 < 0.0).select(0, d1).matrix().squaredNorm();
 }
 //-----------------------------------------------------------------------------
-std::pair<int, double> geometry::compute_closest_entity(
-    const BoundingBoxTree& tree, const BoundingBoxTree& tree_midpoint,
-    const Eigen::Vector3d& p, const mesh::Mesh& mesh)
-{
-  // Closest entity only implemented for cells. Consider extending this.
-  if (tree.tdim() != mesh.topology().dim())
-  {
-    throw std::runtime_error("Cannot compute closest entity of point. "
-                             "Closest-entity is only implemented for cells");
-  }
-
-  // Search point cloud to get a good starting guess
-  const auto [index0, distance0] = compute_closest_point(tree_midpoint, p);
-
-  // Return if we have found the point
-  if (distance0 == 0.0)
-    return {index0, distance0};
-
-  // Call recursive find function
-  const auto [index1, distance1] = _compute_closest_entity(
-      tree, p, tree.num_bboxes() - 1, mesh, index0, distance0 * distance0);
-  assert(index1 >= 0);
-
-  return {index1, std::sqrt(distance1)};
-}
-//-----------------------------------------------------------------------------
 std::pair<int, double>
-geometry::compute_closest_point(const BoundingBoxTree& tree,
-                                const Eigen::Vector3d& p)
+geometry::compute_closest_entity(const BoundingBoxTree& tree,
+                                 const Eigen::Vector3d& p,
+                                 const mesh::Mesh& mesh, double R)
 {
-  // Closest point only implemented for point cloud
-  if (tree.tdim() != 0)
+  // If bounding box tree is empty (on this processor) end search
+  if (tree.num_bboxes() == 0)
+    return {-1, -1};
+
+  // If initial search radius is 0 we estimate the initial distance to the point
+  // using a "random" node from the tree.
+  std::int32_t initial_guess;
+  if (R < 0)
   {
-    throw std::runtime_error("Cannot compute closest point. "
-                             "Search tree has not been built for point cloud");
+    initial_guess = 0;
+    R = std::sqrt((tree.get_bbox(initial_guess).row(0).transpose().matrix() - p)
+                      .squaredNorm());
```

Comment:
Why take sqrt? (could just use `.norm()` method). But why not just keep R^2, it is cheaper...

Reply:
Maybe just add a new variable
`const double R2 = (R < 0) ? tree.get_bbox(...).squaredNorm() : R*R; `